opensearch-operator/pkg/builders/cluster_test.go

```diff
@@ -251,6 +252,39 @@ var _ = Describe("Builders", func() {
 		})
 	})
 
+	When("When Reconciling the snapshotRepoJob: Check container", func() {
+		It("should create a snapshotconfig batch job", func() {
+			clusterObject := ClusterDescWithVersion("2.2.1")
+
+			var snapshotRepoSettings = map[string]string{"bucket": "opensearch-s3-snapshot", "region": "us-east-1", "base_path": "os-snapshot"}
+			snapshotConfig := opsterv1.SnapshotRepoConfig{
+				Name:     "os-snap",
+				Type:     "s3",
+				Settings: snapshotRepoSettings,
+			}
+			clusterObject.Spec.General.SnapshotRepositories = []opsterv1.SnapshotRepoConfig{snapshotConfig}
+			result := NewSnapshotRepoconfigUpdateJob(&clusterObject, "snapshotrepoconfig", "foobar", "snapshotrepoconfig/checksum", nil, nil)
+			Expect(result.Spec.Template.Spec.Containers[0].Name).To(Equal("snapshotrepoconfig"))
+		})
+	})
+
+	When("When Reconciling the snapshotRepoJob: Check Config", func() {
+		It("should create a snapshotconfig batch job", func() {
```

Comment:
This It block and the one above should be merged into one as you are just checking different properties of the update job and you don't need to duplicate the input parameters.